### PR TITLE
fix(clerk-js): Smoother ImpersonationFab animation

### PR DESCRIPTION
**Pull request by desiprisg**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Make the impersonation fab max-width animation smoother by passing a ch value.

### CODE REVIEW

File: packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx [hunk 1]
```diff
@@ -101,6 +102,11 @@ export const ImpersonationFab = () => {
     return null;
   }
 
+  const title = localizationKeys('impersonationFab.title', {
+    identifier: getFullName(session.user) || getIdentifier(session.user),
+  });
+  const titleLength = useLocalizations().t(title).length;
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx:108]: 🛑 This is against [the rules of hooks](https://reactjs.org/docs/hooks-rules.html), as you should not have a hook conditionally executed - line 102 includes an early return statement. Let's move it near the top of the component definition: 

```
  const { t } = useLocalizations();
...
  if (!isImpersonating || !session.user) {
    return null;
  }
...
  const titleLength = t(title).length;
```

Can you check if eslint throws an error/warning for this one here? Thanks!

File: packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx [hunk 2]
```diff
@@ -117,9 +123,9 @@ export const ImpersonationFab = () => {
           backgroundColor: t.colors.$white,
           fontFamily: t.fonts.$main,
           ':hover #cl-impersonationText': {
-            maxWidth: `calc(50vw - ${eyeWidth} - 2 * ${right})`,
+            maxWidth: `min(calc(50vw - ${eyeWidth} - 2 * ${right}), ${titleLength}ch)`,
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx:126]: I was pretty much sure you couldn't live with the delayed animation - nicely done ✔️

**Review by nikosdouvlis — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx (hunk 1 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx:108]: Right! My bad! I missed that.

Comment (desiprisg) [reply packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx]: ✔️ fixed

**Review by nikosdouvlis — APPROVED** [review]
Very nice ✔️

**Pull request merged**